Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -179,18 +179,32 @@ extension NavigationMapView {
             return
         }
         
-        do {
-            try mapView.mapboxMap.style.updateLayer(withId: mainRouteLayerIdentifier) { (lineLayer: inout LineLayer) throws in
-                let mainRouteLayerGradient = self.updateRouteLineGradientStops(fractionTraveled: fractionTraveled, gradientStops: currentLineGradientStops)
-                lineLayer.lineGradient = .expression(Expression.routeLineGradientExpression(mainRouteLayerGradient, lineBaseColor: trafficUnknownColor))
+        let mainRouteLayerGradient = self.updateRouteLineGradientStops(fractionTraveled: fractionTraveled, gradientStops: currentLineGradientStops)
+        let mainRouteLayerGradientExpression = Expression.routeLineGradientExpression(mainRouteLayerGradient, lineBaseColor: trafficUnknownColor)
+        
+        if let data = try? JSONEncoder().encode(mainRouteLayerGradientExpression.self),
+           let jsonObject = try? JSONSerialization.jsonObject(with: data, options: []) {
+            do {
+                try mapView.mapboxMap.style.setLayerProperty(for: mainRouteLayerIdentifier,
+                                                             property: "line-gradient",
+                                                             value: jsonObject)
```

Comment:
Working with raw JSON here is really unfortunate. Client code shouldn’t have to know about the implementation details underpinning the `LineLayer.lineGradient` property.

Reply:
There's a [ticket](https://github.com/mapbox/mapbox-maps-ios/issues/659) tracking some other ways to update the `Expression` property in Map side.